cli/QuickstepCli.cpp

```diff
@@ -323,13 +326,20 @@ int main(int argc, char* argv[]) {
                                    worker_client_ids,
                                    worker_numa_nodes);
 
-  foreman.setWorkerDirectory(&worker_directory);
+  Foreman foreman(main_thread_client_id,
+                  &worker_directory,
+                  &bus,
+                  query_processor->getDefaultDatabase(),
+                  query_processor->getStorageManager(),
+                  num_numa_nodes_system);
 
   // Start the worker threads.
   for (Worker &worker : workers) {
     worker.start();
   }
 
+  foreman.start();
```

Comment:
In the distributed version, `Foreman` starts first, and then `Worker`s starts to discover `Foreman`.

So, I believe this is the right flow for the single node as well, although in this case, `Foreman` knows `Worker`s in advance, and I admit that it does not affect the correctness. But to me it is the best practice to initialize and terminate threads.

@pateljm Any comments?